cuqi/distribution/_joint_distribution.py

```diff
@@ -201,15 +205,24 @@ def _reduce_to_single_density(self):
             # Ensure parameter names match, otherwise return the joint distribution
             if set(self._likelihoods[0].get_parameter_names()) != set(self._distributions[0].get_parameter_names()):
                 return self
-            return Posterior(self._likelihoods[0], self._distributions[0])
-        
+            return self._add_constants_to_density(Posterior(self._likelihoods[0], self._distributions[0]))
+
         # If exactly one distribution and no likelihoods its a Distribution
         if n_dist == 1 and n_likelihood == 0:
-            return self._distributions[0]
-
+            return self._add_constants_to_density(self._distributions[0])        
+        
         # If no distributions and exactly one likelihood its a Likelihood
         if n_likelihood == 1 and n_dist == 0:
             return self._likelihoods[0]
+        
+    def _add_constants_to_density(self, density: Density):
+        """ Add the constants (evaluated densities) to a single density. Used when reducing to single density. """
+        constant = self._sum_evaluated_densities()
+        if isinstance(density, EvaluatedDensity):
+            density.value += constant
```

Comment:
just to double-check here, would that add the constant of the evaluated density twice? could we just do `density.value = constant`? without `+=`. 

Not sure though

Reply:
No. It is because the EvaluatedDensity is a bit a special case. It essentially represents a constant when e.g. we evaluate a logd using all variables. It uses the attribute .value to store the constant rather than constant. Perhaps we can update that in another PR if we want.

Reply:
Thank you @nabriis, I meant here the `+=` instead of `=`. Because it seemed to me that self._sum_evaluated_densities() will already count the constant in `density.value`, so it seemed to me we are incorporating it twice. But still not sure

Reply:
This was a good catch @amal-ghamdi. This code path was actually never used, but may potentially have been in the future. I added a guarding statement to avoid accidentally attempting to add constants to the evaluated density in the future. Your question also made my find the unused case where one does as follows:
```python
import cuqi
d = cuqi.distribution.Gamma(1, 1)
s = cuqi.distribution.Gamma(1, 1)
x = cuqi.distribution.Gaussian(0, 1)
J = cuqi.distribution.JointDistribution(d,s,x)
J(d=1, s=1, x=1)
```
This currently returns "None", but should return sometihng that represents a constant. I fixed that also.

Reply:
👍